niceplots/utils.py

```diff
@@ -1,127 +1,109 @@
-        # Set background to set_dark_background
-        plt.rcParams["axes.facecolor"] = set_dark_background
-        plt.rcParams["figure.facecolor"] = set_dark_background
-        plt.rcParams["patch.edgecolor"] = set_dark_background
-    else:
-        raise TypeError(f"set_dark_background value of {set_dark_background} is invalid")
-
-
-def get_niceColors():
-    # Define an ordered dictionary of some nice Doumont style colors to use as the default color cycle
-    niceColors = OrderedDict()
-    niceColors["Yellow"] = "#e29400ff"  # '#f8a30dff'
-    niceColors["Blue"] = "#1E90FF"
-    niceColors["Red"] = "#E21A1A"
-    niceColors["Green"] = "#00a650ff"
-    niceColors["Maroon"] = "#800000ff"
-    niceColors["Orange"] = "#ff8f00"
-    niceColors["Purple"] = "#800080ff"
-    niceColors["Cyan"] = "#00A6D6"
-    niceColors["Black"] = "#000000ff"
-    # The 2 colours below are not used in the colour cycle as they are too close to the other colours.
-    # Grey is kept in the dictionary as it is used as the default axis/tick colour.
-    # RedOrange is the old Orange, and is kept because I think it looks nice.
-    niceColors["Grey"] = "#5a5758ff"
-    niceColors["RedOrange"] = "#E21A1A"
-
-    return niceColors
-
-
-def get_delftColors():
-    # Define an ordered dictionary of the official TU Delft colors to use as the default color cycle
-    delftColors = OrderedDict()
-    delftColors["Cyan"] = "#00A6D6"  # '#f8a30dff'
-    delftColors["Yellow"] = "#E1C400"
-    delftColors["Purple"] = "#6D177F"
-    delftColors["Red"] = "#E21A1A"
-    delftColors["Green"] = "#A5CA1A"
-    delftColors["Blue"] = "#1D1C73"
-    delftColors["Orange"] = "#E64616"
-    delftColors["Grey"] = "#5a5758ff"
-    delftColors["Black"] = "#000000ff"
-
-    return delftColors
+    # If the style is a niceplots style, return the file path
+    if styleName in get_available_styles():
+        curDir = os.path.dirname(os.path.abspath(__file__))
+        return os.path.join(curDir, "styles", styleName + ".mplstyle")
+
+    # Otherwise assume it's a matplotlib style and just return the style name
+    return styleName
+
+
+def get_colors():
+    """
+    Get a dictionary with the colors for the current style. This function
+    only works when niceplots styles are used (not built-in matplotlib ones).
+
+    Returns
+    -------
+    dict
+        Dictionary of the colors for the requested style. The keys
+        are human-readable names and the keys are the hex codes.
+    """
+    # Get the color codes and their names from the (hopefully) "special" parameter
+    color_codes = get_colors_list()
+    color_names = plt.rcParams["keymap.help"]
+
+    # Ensure that the amount of color names matches the amount of colors
+    assert len(color_codes) == len(
+        color_names
+    ), "The colors are not properly named in the stylesheet, please open an issue on GitHub with the details!"
+
+    # Make the dictionary and return it
+    return dict(zip(color_names, color_codes))
```

Comment:
Do we want to make this an ordered dict to also preserve the order in the colour cycle?